src/main/java/core/basesyntax/operation/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        Storage.storage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
```

Comment:
Handling Purchase operation. According to the checklist, when processing a 'BALANCE' operation, you should check that the result balance is not negative. This implementation directly puts the quantity into the storage without any checks. Consider validating the quantity to ensure it is not negative before updating the storage.